### PR TITLE
feat(validation): returns output from validate function

### LINKED ISSUE #4797: Feature request: `validate()` to return validated payload
### Use case

The Validation feature in Powertools uses `fastjsonschema.validate()` to perform the validation.  This function in `fastjsonschema` returns the validated payload.  While usually this is exactly the same as the input payload, `fastjsonschema` also populates `default` values in missing fields.  For example, with this schema:

```
{
  "properties": {
   "required_id": {
    "type": "integer"
   },
   "test_default": {
    "type": "string",
    "default": "def"
   }
  },
  "required": [
   "required_id"
  ],
  "type": "object"
}
```
And this payload:
```
{
  "required_id": 1234
}
```

The `validate()` function will return:

```
{
  "required_id": 1234,
  "test_default": "def"
}
```

### Solution/User Experience

I'm wondering if we can have the Powertools `validate()` function return the results from `fastjsonschema.validate()`.

### Alternative solutions

_No response_

### Acknowledgment

- [X] This feature request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by dracozombie19**
**Issue number:** #4797

### Changes

The Validation feature has been updated to return the output of the `fastjsonschema.validate()` function.  When validating the payload, `fastjsonschema` returns the payload with `default` values populated where applicable.

### User experience

Consider this line of code:

```python
from aws_lambda_powertools.utilities.validation import validate

val_payload = validate(payload, schema)
```

Before the change, `val_payload` would be `None`.  After the change, it would be the original `payload` value, with any `default` values inserted for fields that were not provided.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change? - No</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: aws_lambda_powertools/utilities/validation/base.py [hunk 1]
```diff
@@ -31,6 +31,12 @@ def validate_data_against_schema(
         Arguments that will be passed directly to the underlying validation call, in this case fastjsonchema.validate.
         For all supported arguments see: https://horejsek.github.io/python-fastjsonschema/#fastjsonschema.validate
 
+    Returns
+    -------
+    Dict
+        The validated event.  If the schema includes a `default` for fields that are not provided, they will be included
+        in the response.
```
Comment (leandrodamascena) [inline aws_lambda_powertools/utilities/validation/base.py]: Nice one! We love docstring ❤️

File: tests/functional/validator/conftest.py [hunk 2]
```diff
@@ -30,6 +30,34 @@ def schema():
     }
 
 
+@pytest.fixture
+def schema_default():
+    return {
+        "$schema": "http://json-schema.org/draft-07/schema",
+        "$id": "http://example.com/example.json",
```
Comment (leandrodamascena) [inline tests/functional/validator/conftest.py]: Sonarcloud is [complaining](https://sonarcloud.io/project/security_hotspots?id=aws-powertools_powertools-lambda-python&pullRequest=4839&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) about security here. Can you please make this endpoint https?

 ```suggestion
        "$schema": "https://json-schema.org/draft-07/schema",
        "$id": "https://example.com/example.json",
```

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Hey @dracozombie19! Thanks a lot for investing time in this PR, returning the validate schema and including the payload is really helpful! Overall this look great, just a small thing about Sonarcloud but it's easy to fix.

One thing that is missing here is to update our documentation with this new behavior because we're now returning the JSON. I'll add an example in our documentation and I expect to merge this PR on Monday.

Great work! 🚀

**Comment by leandrodamascena** [comment]
Hey @sthulb I made some changes in this PR and I need your review here, pls.

**Review by sthulb — APPROVED** [review]

**Review by leandrodamascena — APPROVED** [review]

**Pull request merged**